### PR TITLE
chore: Notify user of custom label colors and related Dashboard color scheme

### LINKED ISSUE #17318: [Explore] Linear color scheme options do not work
While scrolling the Linear color scheme options, suddenly all the options become the same.

#### How to reproduce the bug

1. Go to the "Vaccine Candidates per Country & Stage" sample Chart
2. Click on "Linear color scheme" under "Heatmap options"
3. Scroll the options to the bottom and up a few times
4. Suddenly all the options become the same

### Expected results

While scrolling the color scheme options should not become the same

### Actual results

All color scheme options suddenly become the same while scrolling

#### Screenshots

https://user-images.githubusercontent.com/60598000/139827865-63f1b97b-96a6-4436-b608-bcfe6fb27575.mp4

### Environment

(please complete the following information):

- browser type and version: Chrome 95.0.4638.54
- superset version: master

### Checklist

Make sure to follow these steps before submitting your issue - thank you!

- [ ] I have checked the superset logs for python stacktraces and included it here as text if there are any.
- [x] I have reproduced the issue with at least the latest released version of superset.
- [x] I have checked the issue tracker for the same issue and I haven't found one similar.

### PR DESCRIPTION
**Pull request by geido**
### SUMMARY

This PR: 

- Adds the ability to notify the user about custom label colors used in a Dashboard
- Adds the ability to inform the user when a chart in Explore is having a dashboard context, thus the color scheme will be inherited from the dashboard.
- Adds some test cases for the ColorSchemeControl which had none before
- Fixes #17318

### SUPERSET-UI PR
Requires https://github.com/apache-superset/superset-ui/pull/1467 to be merged to function fully

### AFTER DASHBOARD

https://user-images.githubusercontent.com/60598000/141459740-af81102e-c468-4fd7-855d-535c5c280804.mp4

### AFTER EXPLORE

https://user-images.githubusercontent.com/60598000/141459760-df5ed5b2-5888-4554-bb79-b116f7b1e5e3.mp4

### TESTING INSTRUCTIONS
1. Open a Dashboard
2. Set a color scheme
3. Add custom label colors
4. Make sure the color scheme alerts you of custom label colors in use
4. Visit a chart
5. Make sure the color scheme cannot be changed in Explore

### ADDITIONAL INFORMATION

- [x] Has associated issue: #17318
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by amitmiran137** [comment]
This is great 👌

File: superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx [hunk 1]
```diff
@@ -101,43 +115,93 @@ export default class ColorSchemeControl extends React.PureComponent {
...
+
+      options = filteredColorOptions.map(([value]) => ({
+        customLabel: this.renderOption(value),
+        label: this.schemes?.[value]?.label || value,
+        value,
+      }));
+
+      currentScheme =
+        this.props.value ||
+        (this.props.default !== undefined ? this.props.default : undefined);
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx:182]: Can we simplify that to `this.props.value || this.props.default`?

**Review by kgabryje — APPROVED** [review]
1 suggestion, looks great!

File: superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx (hunk 1 above)
Comment (geido) [reply superset-frontend/src/explore/components/controls/ColorSchemeControl/index.jsx:182]: `default` could also be `null` and `null` it's an acceptable value for the Select. That's the reason of the strict check here

**Pull request merged**